src/field.cpp

```diff
@@ -152,6 +153,13 @@ void field_entry::do_decay()
     // Bypass set_field_age() so we don't reset decay_time;
     age += 1_turns;
     if( type.obj().half_life > 0_turns && get_field_age() > 0_turns ) {
+        // Legacy handling for fire because it's weird and complicated.
+        if( get_field_type() == STATIC( field_type_str_id( "fd_fire" ) ) ) {
```

Comment:
This isn't necessary or anything, but there's no strict need to use the function.
```suggestion
        if( type == STATIC( field_type_str_id( "fd_fire" ) ) ) {
```